pkg/storage/stores/shipper/indexshipper/indexgateway/gateway.go

```diff
@@ -221,17 +221,25 @@ func (g *Gateway) GetChunkRef(ctx context.Context, req *logproto.GetChunkRefRequ
 
 	initialChunkCount := len(result.Refs)
 
-	// Return unfiltered results if there is no bloom querier (Bloom Gateway disabled) or if there are not filters.
-	if g.bloomQuerier == nil || len(req.Filters) == 0 {
-		level.Info(g.log).Log("msg", "chunk filtering is not enabled or there is no line filter", "filters", len(req.Filters))
+	// Return unfiltered results if there is no bloom querier (Bloom Gateway disabled)
+	if g.bloomQuerier == nil {
+		level.Info(g.log).Log("msg", "chunk filtering is not enabled")
+		return result, nil
+	}
+
+	// Extract LineFiltersExpr from the plan. If there is none, we can short-circuit and return before making a req
+	// to the bloom-gateway (through the g.bloomQuerier)
+	filters := req.Plan.LineFilterExprs()
```

Comment:
Here you can use the visitor to extract the line filters, instead of relying on `Plan.LineFilterExprs()`